XDMA/linux-kernel/xdma/alinx_arch.h

```diff
@@ -111,10 +115,8 @@ struct qav_state {
 };
 
 struct buffer_tracker {
-	sysclock_t free_at[HW_QUEUE_SIZE];
-	int head; // Insert
-	int tail; // Remove
-	int count;
+	uint32_t entry_count;
+	uint32_t last_tx_count;
```

Comment:
I think 32 bit is not enough. If we send 1000 packet per second. it longs 50 days which is too short for servers.